src/fe_utils/frontend_hook.c

```diff
@@ -0,0 +1,166 @@
+#include "fe_utils/frontend_hook.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <stdbool.h>
+
+#include <dlfcn.h>    // for dlopen()
+#include <sys/stat.h> // for stat()
+
+#include "c.h"
+#include "port.h"
+#include "pg_config.h"
+
+#define MAXPATHLEN 256 // in src/port/glob.c
+
+char FrontednHookPgInstallPath[MAXPATHLEN] = "\0", FrontendHookPgDataPath[MAXPATHLEN] = "\0";
+
+typedef struct df_files {
+	struct df_files *next;    /* List link */
```

Comment:
There's a simple linked list implementation in `fe_utils/simple_list.h`, Shall we use that one?

Reply:
actually, this is as same as `src/backend/utils/fmgr/dfmgr.c`